Posting empty file as SSL certificate returns HTTP 500

Related #474 
>If you can find any case where you can force a 500 to occur in an integration test then I would say that should actually be filed as a bug

This should return an HTTP `400` instead of `500`.
```
$ curl -F cert=@empty.cer https://0.0.0.0:8181/api/v2/certificates/
{"meta":{"type":"text/plain","status":"Internal Server Error"},"data":{"reason":"Internal Server Error"}}
```
The cryostat logs show that an Exception was thrown and it was not of type `IOException` or `CertificateEncodingException`. The `rootCauseMessage()` is also `Internal Server Error`.
```
SEVERE: Exception thrown
io.cryostat.net.web.http.api.v2.ApiException: Internal Server Error
Caused by: io.cryostat.net.web.http.api.v2.ApiException: Internal Server Error
```

--- a/src/main/java/io/cryostat/net/web/http/api/v2/CertificatePostHandler.java
+++ b/src/main/java/io/cryostat/net/web/http/api/v2/CertificatePostHandler.java
@@ -38,6 +38,7 @@
 package io.cryostat.net.web.http.api.v2;
 
 import java.io.File;
+import java.io.FileNotFoundException;
 import java.io.FileOutputStream;
 import java.io.IOException;
 import java.io.InputStream;
@@ -154,7 +155,7 @@ class CertificatePostHandler extends AbstractV2RequestHandler<Path> {
             Collection<? extends Certificate> certificates = certValidator.parseCertificates(fis);
 
             if (certificates.isEmpty()) {
-                throw new ApiException(500, "No certificates found");
+                throw new FileNotFoundException("No certificates found");
             }
 
             try (FileOutputStream out = outputStreamFunction.apply(filePath.toFile())) {
@@ -165,6 +166,8 @@ class CertificatePostHandler extends AbstractV2RequestHandler<Path> {
                 }
             }
 
+        } catch (FileNotFoundException e) {
+            throw new ApiException(400, e.getMessage(), e);
         } catch (IOException ioe) {
             throw new ApiException(500, ioe.getMessage(), ioe);
         } catch (CertificateEncodingException cee) {

--- a/src/test/java/itest/UploadCertificateIT.java
+++ b/src/test/java/itest/UploadCertificateIT.java
@@ -38,15 +38,21 @@
 package itest;
 
 import java.io.File;
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
 import java.util.concurrent.TimeUnit;
 
 import io.vertx.core.buffer.Buffer;
+import io.vertx.core.json.JsonObject;
 import io.vertx.ext.web.client.HttpResponse;
 import io.vertx.ext.web.multipart.MultipartForm;
 import itest.bases.StandardSelfTest;
+import io.vertx.ext.web.handler.impl.HttpStatusException;
+
 import org.hamcrest.MatcherAssert;
 import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 
 public class UploadCertificateIT extends StandardSelfTest {
@@ -55,6 +61,7 @@ public class UploadCertificateIT extends StandardSelfTest {
     static final String FILE_NAME = "empty.cer";
     static final String TRUSTSTORE_CERT = "truststore/" + FILE_NAME;
     static final String MEDIA_TYPE = "application/pkix-cert";
+    static final String REQ_URL = String.format("/api/v2/certificates");
 
     @Test
     public void shouldNotAddEmptyCertToTrustStore() throws Exception {
@@ -70,7 +77,7 @@ public class UploadCertificateIT extends StandardSelfTest {
                         .binaryFileUpload(CERT_NAME, FILE_NAME, path, MEDIA_TYPE);
 
         webClient
-                .post(String.format("/api/v2/certificates"))
+                .post(REQ_URL)
                 .sendMultipartForm(
                         form,
                         ar -> {
@@ -85,5 +92,23 @@ public class UploadCertificateIT extends StandardSelfTest {
         int statusCode = uploadRespFuture.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
 
         MatcherAssert.assertThat(statusCode, Matchers.equalTo(500));
+    }
+
+    @Test
+    public void shouldThrowWhenPostingWithoutCert() throws Exception {
+
+        CompletableFuture<JsonObject> response = new CompletableFuture<>();
+
+        webClient
+                .post(REQ_URL)
+                .send(
+                        ar -> {
+                            assertRequestStatus(ar, response);
+                        });
+        ExecutionException ex =
+                Assertions.assertThrows(ExecutionException.class, () -> response.get());
+        MatcherAssert.assertThat(
+                ((HttpStatusException) ex.getCause()).getStatusCode(), Matchers.equalTo(400));
+        MatcherAssert.assertThat(ex.getCause().getMessage(), Matchers.equalTo("Bad Request"));
     }
 }

--- a/src/test/java/itest/UploadCertificateIT.java
+++ b/src/test/java/itest/UploadCertificateIT.java
@@ -38,18 +38,13 @@
 package itest;
 
 import java.io.File;
-import java.util.Map;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ExecutionException;
-import java.util.concurrent.TimeUnit;
 
-import io.vertx.core.buffer.Buffer;
 import io.vertx.core.json.JsonObject;
-import io.vertx.ext.web.client.HttpResponse;
+import io.vertx.ext.web.handler.impl.HttpStatusException;
 import io.vertx.ext.web.multipart.MultipartForm;
 import itest.bases.StandardSelfTest;
-import io.vertx.ext.web.handler.impl.HttpStatusException;
-
 import org.hamcrest.MatcherAssert;
 import org.hamcrest.Matchers;
 import org.junit.jupiter.api.Assertions;
@@ -66,7 +61,7 @@ public class UploadCertificateIT extends StandardSelfTest {
     @Test
     public void shouldNotAddEmptyCertToTrustStore() throws Exception {
 
-        CompletableFuture<Integer> uploadRespFuture = new CompletableFuture<>();
+        CompletableFuture<Integer> response = new CompletableFuture<>();
         ClassLoader classLoader = getClass().getClassLoader();
         File emptyCert = new File(classLoader.getResource(FILE_NAME).getFile());
         String path = emptyCert.getAbsolutePath();
@@ -81,17 +76,13 @@ public class UploadCertificateIT extends StandardSelfTest {
                 .sendMultipartForm(
                         form,
                         ar -> {
-                            if (ar.failed()) {
-                                uploadRespFuture.completeExceptionally(ar.cause());
-                                return;
-                            }
-                            HttpResponse<Buffer> result = ar.result();
-                            uploadRespFuture.complete(result.statusCode());
+                            assertRequestStatus(ar, response);
                         });
-
-        int statusCode = uploadRespFuture.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
-
-        MatcherAssert.assertThat(statusCode, Matchers.equalTo(500));
+        ExecutionException ex =
+                Assertions.assertThrows(ExecutionException.class, () -> response.get());
+        MatcherAssert.assertThat(
+                ((HttpStatusException) ex.getCause()).getStatusCode(), Matchers.equalTo(400));
+        MatcherAssert.assertThat(ex.getCause().getMessage(), Matchers.equalTo("Bad Request"));
     }
 
     @Test

